DOCS-538: Refactor SCUTTLE gamepad tutorial (#1545)
viamrobotics/docs: docs/tutorials/control/scuttle-gamepad.md

{{< tabs >}}
{{% tab name="Config Builder" %}}

- Go to the **Services** subtab of your robot's **Config** tab.
- In the **Create service** panel, click the **Type** dropdown and select `Base Remote Control`.
- Enter `gamepad_service` for the **Service** **name**.
- Click **Create service**.

Copy and paste the following into the empty **Attributes** field, replacing `<your-base-name>` with your base's name.

```json {class="line-numbers linkable-line-numbers"}
{
 "base": "<your-base-name>",
 "input_controller": "my-gamepad"
}
```

<br>

![Service configuration](/tutorials/scuttle-gamepad/gamepad-service-config.png)

{{% /tab %}}
{{% tab name="Raw JSON" %}}

If instead of using the config builder, you prefer to write raw JSON, switch to **Raw JSON** mode on the **Config** tab.
Add the following configuration for your base remote control service, replacing `<your-base-name>` with your base's name:

```json {class="line-numbers linkable-line-numbers"}
  "services": [
    {
      "name": "gamepad_service",
      "type": "base_remote_control",
      "attributes": {
        "input_controller": "my-gamepad",
        "base": "<your-base-name>"
      }
    }
  ]

> ```

Author: ```suggestion
```

If you already have a `"services"` array with other services configured, add just the contents of the square brackets above to that array, rather than creating two different `"services"` arrays.
```

> 

Author: ```suggestion

If you already have a `"services"` array with other services configured, add just the contents of the square brackets above to that array, rather than creating two different `"services"` arrays.
```

cost: 575
---

This tutorial teaches you how to add a Bluetooth dongle gamepad controller to your wheeled robot.
By the end of this tutorial, you'll be able to drive your rover around like an RC car.

{{% alert title="Tip" color="tip" %}}

If your gamepad has a dongle, keep reading.
If your gamepad does not have a dongle, check out [Drive a Yahboom Rover with a Gamepad](../yahboom-rover/#connecting-a-bluetooth-controller) for Bluetooth pairing instructions.

{{% /alert %}}

## Requirements

You will need the following to complete this tutorial:

> - A wheeled rover, configured with a [base component](/components/base/) on the [Viam app](https://app.viam.com/)

Author: ```suggestion
- A wheeled rover, configured with a [base component](/components/base/) on the [Viam app](https://app.viam.com/).
  This tutorial uses a [SCUTTLE rover](https://www.scuttlerobot.org/shop/) as an example but you can complete this tutorial using a different rover.
```

  - [Setting up a SCUTTLE with Viam](/tutorials/configure/scuttlebot/) is a good place to start if you haven't already configured your base.

>     This tutorial uses a [SCUTTLE rover](https://www.scuttlerobot.org/shop/) as an example but you complete this tutorial using a different rover.

Author: ```suggestion
```

- [EasySMX ESM-9101 Wireless Controller](https://droix.net/product/easysmx-esm-9101/) or a similar gamepad and dongle.
  This is the controller that comes with the SCUTTLE rover.

{{<video webm_src="/tutorials/videos/scuttledemos_gamepad.webm" mp4_src="/tutorials/videos/scuttledemos_gamepad.mp4" alt="Controlling a SCUTTLE Robot using a Bluetooth gamepad" poster="/tutorials/scuttlebot/scuttledemos_gamepad.jpg">}}

## Set up the hardware

Plug the gamepad Bluetooth dongle into a USB port on the rover's [board](/components/board/).
Turn on power to the rover.

## Add the controller to the rover's config

Go to your rover's **Config** tab on the [Viam app](https://app.viam.com/).

In the **Create component** panel, configure a [gamepad](/components/input-controller/gamepad/):

- Enter `my-gamepad` as the component **Name**.
- Select `input_controller` for the component **Type**.
- Select `gamepad` for the **Model**.
- Click **Create component**.

![Blank configuration JSON](/tutorials/scuttle-gamepad/gamepad-config.png)

The controller config adds the gamepad controller to your robot.
However, it does not connect it to any functionality.
This requires a Service.

> ## Add the Base Remote Control Service

Reviewer: Again also please add a raw json option

Services are software packages that provide robots with higher level functionality.
To link the controller's input to the base functionality, you need to configure the [Base Remote Control Service](/services/base-rc/):

- Go to the **Services** subtab of your robot's **Config** tab.
- In the **Create service** panel, click the **Type** dropdown and select `Base Remote Control`.
- Enter `gamepad_service` for the **Service** **name**.
- Click **Create service**.

> Copy and paste the following into the empty **Attributes** field, replacing `THE NAME OF YOUR BASE` with your base's name.

Author: ```suggestion
Copy and paste the following into the empty **Attributes** field, replacing `<your-base-name>` with your base's name.
```

```json {class="line-numbers linkable-line-numbers"}
{

>  "base": "THE NAME OF YOUR BASE",

Reviewer: Lets follow the same style as in the components docs
```suggestion
 "base": "<your-base-name>",
```

 "input_controller": "my-gamepad"
}
```

<br>

![Service configuration](/tutorials/scuttle-gamepad/gamepad-service-config.png)

Click **Save config**, then go to the **Control** tab.

You should see the panel for the gamepad and its connection indicator:

![Gamepad input UI showing a "connected" indicator and a list of inputs from all the buttons, for example X=0.0, RY=0.0 and East=0.](/tutorials/scuttle-gamepad/control-tab-input.png)

> At this point, you should be able to move the rover using the gamepad.

Reviewer: Instead of saying "you should be able to", instruct them waht to do to test it.

Author: ```suggestion
Try moving the left joystick or pressing the D-pad to move the rover using the gamepad.
```

> However, whether you can drive it with the joystick or with the D-pad depends on the [settings on the controller](#easysmx-esm-9101-wireless-controller-information).

Author: ```suggestion
```

> If you are in the [mode](#easysmx-esm-9101-wireless-controller-information) that allows you to use the joystick (#7), it will change the `X` and `Y` values on the **Control** tab gamepad panel:

Reviewer: Let's prepend this with info that makes it clear this applies only to that specific controller. Somethiing like:

The ESM9109 controller has two modes that...

Author: ```suggestion
The ESM-9101 controller has different [modes](#easysmx-esm-9101-wireless-controller-information) that allow you to use either the joystick or the D-pad.
If you are in the mode that allows you to use the joystick (#7), it will change the `X` and `Y` values on the **Control** tab gamepad panel:
```

![Blank configuration JSON](/tutorials/scuttle-gamepad/gamepad-config.png)

{{% /tab %}}
{{% tab name="Raw JSON" %}}

> If instead of using the config builder, you prefer to write raw JSON, switch to **Raw JSON** mode on the **Config** tab.
> Inside the `"components": []` section of your config, add the following configuration for your [gamepad](/components/input-controller/gamepad/):

Reviewer: ```suggestion
If instead of using the config builder, you prefer to write raw JSON, switch to [**Raw JSON** mode](/manage/configuration/#the-config-tab) on the **Config** tab.
Inside the `components` array of your config, add the following configuration for your [gamepad](/components/input-controller/gamepad/):
```

![Service configuration](/tutorials/scuttle-gamepad/gamepad-service-config.png)

{{% /tab %}}
{{% tab name="Raw JSON" %}}

> If instead of using the config builder, you prefer to write raw JSON, switch to **Raw JSON** mode on the **Config** tab.
> Add the following configuration for your base remote control service, replacing `<your-base-name>` with your base's name:

Reviewer: ```suggestion
If instead of using the config builder, you prefer to write raw JSON, switch to [**Raw JSON** mode](/manage/configuration/#the-config-tab) on the **Config** tab.
Inside the `services` array of your config, add the following configuration for your base remote control service, replacing `<your-base-name>` with your base's name:
```

Author: ```suggestion
If instead of using the config builder, you prefer to write raw JSON, switch to [**Raw JSON** mode](/manage/configuration/#the-config-tab) on the **Config** tab.
Add the following configuration for your base remote control service, replacing `<your-base-name>` with your base's name:
```
This one is different from the components part, because a new robot in the Viam app autopopulates with just a components array and no services array so unless they've already been doing other things with services on this robot, they don't have a services array to add to yet; they need to make it.
